python/modules/IcePy/Executor.cpp

```diff
@@ -74,14 +74,12 @@ IcePy::initExecutor(PyObject* module)
     return true;
 }
 
-IcePy::Executor::Executor(PyObject* executor) : _executor(executor)
+IcePy::Executor::Executor(PyObject* executor) : _executor(Py_NewRef(executor))
 {
     if (!PyCallable_Check(executor))
     {
         throw Ice::InitializationException(__FILE__, __LINE__, "executor must be a callable");
```

Comment:
Do you need to release _executor in this case? The destructor of Executor won't run if the ctor throws.

Reply:
I don't think so. `_executor` is a PyObjectHandle, its destructor should still be called and it will release the reference.